flink-runtime/src/main/java/org/apache/flink/runtime/operators/util/BloomFilter.java

```diff
@@ -143,4 +176,63 @@ public String toString() {
         output.append(bitSet);
         return output.toString();
     }
+
+    public static byte[] toBytes(BloomFilter filter) {
+        byte[] data = filter.bitSet.getData();
+        int byteSize = data.length;
+        byte[] bytes = new byte[8 + byteSize];
+        UNSAFE.putInt(bytes, BYTE_ARRAY_BASE_OFFSET, filter.numHashFunctions);
+        UNSAFE.putInt(bytes, BYTE_ARRAY_BASE_OFFSET + 4, byteSize);
+        UNSAFE.copyMemory(
+                data, BYTE_ARRAY_BASE_OFFSET, bytes, BYTE_ARRAY_BASE_OFFSET + 8, byteSize);
+        return bytes;
+    }
+
+    public static BloomFilter fromBytes(byte[] bytes) {
+        int numHashFunctions = UNSAFE.getInt(bytes, BYTE_ARRAY_BASE_OFFSET);
+        int byteSize = UNSAFE.getInt(bytes, BYTE_ARRAY_BASE_OFFSET + 4);
+        byte[] data = new byte[byteSize];
+        UNSAFE.copyMemory(
+                bytes, BYTE_ARRAY_BASE_OFFSET + 8, data, BYTE_ARRAY_BASE_OFFSET, byteSize);
+
+        BitSet bitSet = new BitSet(byteSize);
+        bitSet.setMemorySegment(MemorySegmentFactory.wrap(data), 0);
+        return new BloomFilter(bitSet, numHashFunctions);
+    }
+
+    public static byte[] mergeBloomFilterBytes(byte[] bf1Bytes, byte[] bf2Bytes) {
+        return mergeBloomFilterBytes(bf1Bytes, 0, bf1Bytes.length, bf2Bytes, 0, bf2Bytes.length);
+    }
+
+    /** Merge the bf2 bytes to bf1 */
+    public static byte[] mergeBloomFilterBytes(
```

Comment:
can be private

Reply:
Maybe named `mergeSerializedBloomFilters` ? And add doc to indicate that `bf1Bytes` will be changed after the merge.